File: standing-app-ng/src/app/model/team.ts
import { League } from "./league";
import { User } from "./user";

export class Team {
    id: number;
    coachId: number;
    name: string;
    leagueId: number;
    league: League;
    coach: User;

    constructor(
        id: number = 0,
        coachId: number = 0,
        name: string = '',
        leagueId: number = 0,
        league: League = new League(),
        coach: User = new User()
    ) {
        this.id = id;
        this.coachId = coachId;
        this.name = name;
        this.leagueId = leagueId;
        this.league = league;
        this.coach = coach;
    }
}

File: StandingsApp/StandingsApp/Models/EF/Team.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StandingsApp.Models.EF
{
    [Table("Team")]
    public class Team
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CoachId { get; set; }

        [Required]
        [StringLength(75)]
        public string Name { get; set; }

        [Required]
        public int LeagueId { get; set; }

        public virtual User? Coach { get; set; }
        public virtual League? League { get; set; }
    }
}
